src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -36,7 +36,7 @@ internal sealed class PersistentStorageProvider : PartialStorageProviderBase
         private readonly ResettableDictionary<StorageCell, byte[]> _originalValues = new();
 
         private readonly ResettableHashSet<StorageCell> _committedThisRound = new();
-        private readonly Dictionary<StorageCell, byte[]> _blockCache = new(4_096);
+        private readonly Dictionary<Address, Dictionary<StorageCell, byte[]>> _blockCache = new(4_096);
```

Comment:
```suggestion
        private readonly Dictionary<AddressAsKey, Dictionary<StorageCell, byte[]>> _blockCache = new(4_096);
```

Does the second dictionary need to be full `StorageCell` as key or can it be just the `StorageCell.Index`?

Reply:
No idea.